azurerm/internal/services/securitycenter/security_center_server_vulnerability_assessment_resource.go

```diff
@@ -147,17 +147,14 @@ func resourceServerVulnerabilityAssessmentDelete(d *pluginsdk.ResourceData, meta
 		time.Sleep(time.Duration(retry) * time.Second)
 	}
 
-	response, err := client.Delete(ctx, targetId.ResourceGroup, targetId.Provider, targetId.ResourceType, targetId.Name)
+	future, err := client.Delete(ctx, targetId.ResourceGroup, targetId.Provider, targetId.ResourceType, targetId.Name)
 	if err != nil {
-		// https://github.com/Azure/azure-rest-api-specs/issues/12308 the API report error even if delete was successful
-		if utils.ResponseWasStatusCode(response, 202) {
-			return nil
-		} else {
-			return fmt.Errorf("removing Server Vulnerability Assessment for %q: %+v\n", targetId.Id, err)
-		}
-	} else {
-		return nil
+		return fmt.Errorf("deleting Server Vulnerability Assessment %s: %v", targetId.Id, err)
 	}
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of Server Vulnerability Assessment %s: %v", targetId.Id, err)
+	}
+	return nil
```

Comment:
looks like this future doesn't destroy correctly?

```
------- Stdout: -------
=== RUN   TestAccServerVulnerabilityAssessment_basic
=== PAUSE TestAccServerVulnerabilityAssessment_basic
=== CONT  TestAccServerVulnerabilityAssessment_basic
testing_new.go:21: WARNING: destroy failed, so remote objects may still exist and be subject to billing
testing_new.go:21: failed to destroy: exit status 1
Error: waiting for deletion of Server Vulnerability Assessment /subscriptions/*******/resourceGroups/acctestRG-sva-210618065036249376/providers/Microsoft.Compute/virtualMachines/acctestVM-210618065036249376: Future#WaitForCompletion: context has been cancelled: StatusCode=202 -- Original Error: context deadline exceeded
--- FAIL: TestAccServerVulnerabilityAssessment_basic (508.09s)
FAIL
```

Reply:
Turns out needs to extend the timeout value from 5min -> 10min (local run it takes ~7min).

## Test

```bash
💤  TF_ACC=1 go test -timeout=3h -v ./azurerm/internal/services/securitycenter -run="TestAccServerVulnerabilityAssessment_basic"                                                                                                                 
2021/06/18 15:41:17 [DEBUG] not using binary driver name, it's no longer needed
2021/06/18 15:41:17 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccServerVulnerabilityAssessment_basic
=== PAUSE TestAccServerVulnerabilityAssessment_basic
=== CONT  TestAccServerVulnerabilityAssessment_basic
--- PASS: TestAccServerVulnerabilityAssessment_basic (859.07s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/securitycenter      859.265s

```